### PR TITLE
"collection" url mapping inside a "group" behaves the same as it is not in a group. Fixes #10844

### LINKED ISSUE #10844: Resource collection does not work as expected if the resource is within group
### Steps to Reproduce

1. Clone demo app
2. Run url-mapping-report command

### Expected Behaviour

The following url mapping add a URI directly below a resource using collection

```
group '/api', {
    '/employees'(resources: 'employee') {
        collection {
            '/search'(controller: 'employee', action: 'search', method: 'GET')
        }
    }
}
```

Should generate the following result regarding the **search action**

```
Controller: employee
 |   GET    | /api/employees/create                             | Action: create           |
 |   GET    | /api/employees/search                             | Action: search           |
 |   GET    | /api/employees/${id}/edit                         | Action: edit             |
 |   POST   | /api/employees                                    | Action: save             |
 |   GET    | /api/employees                                    | Action: index            |
 |  DELETE  | /api/employees/${id}                              | Action: delete           |
 |  PATCH   | /api/employees/${id}                              | Action: patch            |
 |   PUT    | /api/employees/${id}                              | Action: update           |
 |   GET    | /api/employees/${id}                              | Action: show             |
```

### Actual Behaviour

```
Controller: employee
 |   GET    | /api/employees/create                             | Action: create           |
 |   GET    | /api/employees/${id}/edit                         | Action: edit             |
 |   GET    | /api/employees/${employeeId}/search               | Action: search           |
 |   POST   | /api/employees                                    | Action: save             |
 |   GET    | /api/employees                                    | Action: index            |
 |  DELETE  | /api/employees/${id}                              | Action: delete           |
 |  PATCH   | /api/employees/${id}                              | Action: patch            |
 |   PUT    | /api/employees/${id}                              | Action: update           |
 |   GET    | /api/employees/${id}                              | Action: show             |
```

If I avoid using group the result of using collection within resource is the expected

```
'/api/employees'(resources: 'employee') {
    collection {
        '/search'(controller: 'employee', action: 'search', method: 'GET')
    }
}
```
For the search action I get the following result

`/api/employees/search`

### Environment Information

- **Operating System**: ubuntu 16.04
- **Grails Version:** Using grails version 3.3.1
- **JDK Version:** Using java version 8u144-zulu

### Example Application

https://github.com/ilmoralito/url-mapping-report-demo

### PR DESCRIPTION
**Pull request by ilopmar**
Fixes #10844

### CODE REVIEW

**Comment by jameskleeh** [comment]
Does the original test pass with your changes? It's better to create a new test that covers the behavior or just edit the test enough to get it to pass.

File: grails-web-url-mappings/src/test/groovy/grails/web/mapping/GroupedUrlMappingSpec.groovy [hunk 1]
```diff
@@ -27,30 +27,30 @@ import javax.servlet.http.HttpServletResponse
...
                     }
                 }
             }
         }
 
         expect:
-        linkGenerator.link(controller:'bar', action:'baz', params:[barId:1]) == 'http://localhost/g/bars/1/baz'
+        linkGenerator.link(controller: 'employee1', action: 'search1', method: 'GET', params: [userId: 1]) == 'http://localhost/api1/employees/search?userId=1'
+        linkGenerator.link(controller: 'employee2', action: 'search2', method: 'GET', params: [userId: 1]) == 'http://localhost/api2/employees/search?userId=1'
     }
```
Comment (jameskleeh) [inline grails-web-url-mappings/src/test/groovy/grails/web/mapping/GroupedUrlMappingSpec.groovy]: Perhaps add a test with `employeeId` to verify it puts it in the URL?

**Comment by ilopmar** [comment]
No, the original test doesn't pass because it was a bug and this change is a breaking change, so I though it was better to change the failing test with the new behavior and submit the PR to `master`.

File: grails-web-url-mappings/src/test/groovy/grails/web/mapping/GroupedUrlMappingSpec.groovy (hunk 1 above)
Comment (ilopmar) [reply grails-web-url-mappings/src/test/groovy/grails/web/mapping/GroupedUrlMappingSpec.groovy]: I think that's not possible. According to http://docs.grails.org/3.3.2/guide/theWebLayer.html#restfulMappings in `Nested Resources` section, that mapping is not created automatically.

Comment (ilopmar) [reply grails-web-url-mappings/src/test/groovy/grails/web/mapping/GroupedUrlMappingSpec.groovy]: I give it a second thought and also read the documentation twice and I've push a commit.

**Pull request closed without merging**